src/proj.h

```diff
@@ -73,20 +74,28 @@
  *           as large as the traditional XY and LP objects, timing results
  *           have shown the overhead to be very reasonable.
  *
- *           In its current incarnation, the API is focused on compactness:
- *           Currently it consists of only nine functions.
+ *           Contexts and thread safety
+ *           --------------------------
  *
- *           Hence, due to the proj_ctx subsystem being little used, it has
- *           been replaced by a tiny set of 3 functions, making it possible,
- *           but not very convenient, to do almost everything possible with
- *           the original ctx API.
+ *           After a year of experiments (and previous experience from the
+ *           trlib transformation library) it has become clear that the
+ *           context subsystem is unavoidable in a multi-threaded world.
+ *           Hence, instead of hiding it away, we move it into the limelight,
```

Comment:
@rouault  - The context provides the pj_errno that makes it possible to signal what went wrong in a non-succesful call to proj_create, where we (as in the classic pj_init) return 0 for "non-succesfull).

Yes, it would be nice to just set an errno on the returned object, but since we use a signalling null-pointer, to actually report what went wrong, we need an errno style variable, and we need it to be thread-local in the multithread case.

Hence my suggestion that we use errno, which is thread local on multithreaded systems, and which is intended to signal library call errors. But you argued against both this and explicitly thread local storage class pj_errno a few days ago - which is the reason I had to switch to explicit context handling.

You can't have it both ways - it's either signalling through errno (or a thread local storage class pj_errno) OR explicit contexts.